app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt

```diff
@@ -50,13 +52,28 @@ abstract class TwoPaneFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         setupSlidingPane()
+        observeFolderName()
         observeThreadEvents()
     }
 
     private fun setupSlidingPane() {
         slidingPaneLayout.lockMode = SlidingPaneLayout.LOCK_MODE_LOCKED
     }
 
+    private fun observeFolderName() = with(mainViewModel) {
+
+        currentFolder.observe(viewLifecycleOwner) { folder ->
+
+            val displayedFolder = (if (this@TwoPaneFragment is SearchFragment) {
```

Comment:
If you find an elegant way to to reduce coupling for `this@TwoPaneFragment is SearchFragment` so TwoPaneFragment doesn't need to know about the existence of its possible children, please do